src/jit/codegenxarch.cpp

```diff
@@ -8173,8 +8206,7 @@ CodeGen::genPutStructArgStk(GenTreePtr treeNode, unsigned baseVarNum)
         genConsumePutStructArgStk(putArgStk, REG_RDI, REG_RSI, REG_NA, baseVarNum);
         GenTreePtr   dstAddr = putArgStk;
         GenTreePtr   srcAddr = putArgStk->gtOp.gtOp1;
-        gcInfo.gcMarkRegPtrVal(REG_RSI, srcAddr->TypeGet());
-        gcInfo.gcMarkRegPtrVal(REG_RDI, dstAddr->TypeGet());
+        gcInfo.gcMarkRegPtrVal(REG_RSI, srcAddr->gtGetOp1()->TypeGet());
```

Comment:
Everywhere in this routine what we need is type of op1 of srcAddr.  srcAddr here is actually a gt_ldobj and is not really representing srcAddr.  We can avoid everywhere srcAddr->gtGetOp1() by initializing srcAddr as

GenTreePtr src = putArgStk->gtGetOp1();
assert(src->OperGet() == GT_LDOBJ);
GenTreePtr srcAddr = src->gtGetOp1();